### ISSUE TITLE
LDAP integration

### ISSUE THREAD

**Issue by jonasjoest** [TeaSpeak/TeaSpeak#516]
We use the LDAP protocol in our infrastructure in order to maintain users, groups and roles for all provided services in a centralized manner.

I would love to see a similar feature for TeaSpeak aswell.

Possible affected features:
- server / channel permissions
- musicbot permissions
- whitelist for connections
- etc...

**Comment by WolverinDEV** [comment]
Well I currently don't really understand the advantage behind that enchantment.
Could you explain that a littlebit more?

**Comment by jonasjoest** [comment]
Just to give you a short explanation for what I'm talking about:

In our community we provide some services like file sharing, file cloud, git, issue tracker, ci and other devops tools. All those services are maintained centrally using an LDAP provider. This way we can adjust users and their roles and permissions using one single administration interface. (Even unix permissions, certificate chains and kerberos single-sign on can be set up). That enables us to maintain the access/write/administration rights either globally or per service/feature in our LDAP management interface without touching any configuration file or service-specific admin panel. (Makes things a lot easier for moderators and admins considering many different services in bigger environments)

I could consider different approaches to achieve an LDAP integration to TeaSpeak. Mapping TeaSpeak permissions to LDAP groups is the most usual way in my opinion. (admin/speak power/ music bot power/ etc)
More examples / implementation approaches have to be worked out.

On the other hand you could even manage server configurations via LDAP (i.e. channel structure, channel permissions, channel properties, ports, file sharing, descriptions, ...)


**Just to clarify**: this issue/idea should be more likely considered as "brainstorming" rather than as "this is the way I want you to implement it". I haven't got a clue how this feature could look like in the end.

**To summarize**: I have to admit that this enhancement is IMO not really relevant for most small teams that just want to get their VOIP gaming setup running in a small gaming environment, but I assume that it would come in very handy for bigger communities or server providers with an already existing user-role-concept or bigger service infrastructure. (Maybe this could even be a step towards productive usage of TeaSpeak in development teams.)

**Comment by WolverinDEV** [comment]
Well first of all thanks for your great exploration.
Before you, I've never heard about that, actually IDK how famous this is.

I think to implement and adjust an individual system so its fits into another isnt the best idea in general.
I also think that implementing a system like this is way to much work with much less measurable benefits, but I could be wrong :)

Well okey thats good to know.
Hasnt sound like that in the first place :)

**Comment by WolverinDEV** [comment]
A bit of an update.
I don't think that I'll ever implement LDAP integration.
It does not really fit into the main system how TeaSpeak works.
As well I dont really see a use case in general, well ofc some rare uses may, but thats not worth it.
